lib/realtime/tenants/authorization.ex

```diff
@@ -73,37 +92,69 @@ defmodule Realtime.Tenants.Authorization do
     sub = Map.get(claims, :sub)
     claims = Jason.encode!(claims)
     headers = headers |> Map.new() |> Jason.encode!()
+    channel_name = if channel, do: channel.name, else: nil
 
-    Postgrex.transaction(conn, fn conn ->
-      Postgrex.query(
-        conn,
-        """
-        SELECT
-         set_config('role', $1, true),
-         set_config('realtime.channel_name', $2, true),
-         set_config('request.jwt.claim.role', $3, true),
-         set_config('request.jwt', $4, true),
-         set_config('request.jwt.claim.sub', $5, true),
-         set_config('request.jwt.claims', $6, true),
-         set_config('request.headers', $7, true)
-        """,
-        [role, channel_name, role, jwt, sub, claims, headers]
-      )
-
-      case Postgrex.query(conn, "SELECT name from realtime.channels", [], mode: :savepoint) do
-        {:ok, %{num_rows: 0}} ->
-          {:ok, %Permissions{read: false}}
-
-        {:ok, _} ->
-          {:ok, %Permissions{read: true}}
-
-        {:error, %Postgrex.Error{postgres: %{code: :insufficient_privilege}}} ->
-          {:ok, %Permissions{read: false}}
-
-        {:error, error} ->
-          Logger.error("Error getting permissions for connection: #{inspect(error)}")
-          {:error, error}
-      end
-    end)
+    Postgrex.query(
+      conn,
+      """
+      SELECT
+       set_config('role', $1, true),
+       set_config('realtime.channel_name', $2, true),
+       set_config('request.jwt.claim.role', $3, true),
+       set_config('request.jwt', $4, true),
+       set_config('request.jwt.claim.sub', $5, true),
+       set_config('request.jwt.claims', $6, true),
+       set_config('request.headers', $7, true)
+      """,
+      [role, channel_name, role, jwt, sub, claims, headers]
+    )
+  end
+
+  defp check_read_permissions(conn, permissions) do
+    query = from(c in Channel, select: c.name)
+
+    case Repo.all(conn, query, Channel, mode: :savepoint) do
```

Comment:
We really should start wrapping all these in `rescue`s and return exceptions as error tuples.

Reply:
yes, we can do that at the Repo level so it's simpler 👀 one point of recovery basically

Reply:
Everything that ultimately calls Postgrex.query...

For context for others... because in a multi-tenant env we really have no control over how healthy the database is so we don't want to accidentally crash procs if Postgrex.query throws an exception vs an error tuple.

Reply:
yes yes what I mean is that we can use the extra methods I added to Realtime.Repo to have those rescues wrapped and for most of it the Authorization module will be able to be rescued 👀

Reply:
did the change to capture exists to handle some of this exceptions